modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala

```diff
@@ -181,6 +188,58 @@ final case class Resolver[F[_]](repos: List[Registry], cache: Option[ResolverCac
   ): F[Either[ResolutionError, Json]] =
     lookupSchemaResult(schemaKey).map(_.map(_.value.schema))
 
+  def lookupSchemasUntil(
+    maxSchemaKey: SchemaKey
+  )(implicit
+    F: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): F[Either[SchemaResolutionError, List[RawSchema]]] = {
+
+    // If Iglu Central or any of its mirrors doesn't have a schema,
+    // it should be considered NotFound, even if one of them returned an error
+    def isNotFound(error: ResolutionError): Boolean = {
+      val (igluCentral, custom) = error.value.partition { case (repo, _) =>
+        allIgluCentral.contains(repo)
+      }
+      custom.values.flatMap(_.errors).forall(_ == RegistryError.NotFound) &&
+      (igluCentral.isEmpty || igluCentral.values.exists(
+        _.errors.forall(_ == RegistryError.NotFound)
+      ))
+    }
```

Comment:
Hi @istreeter , what do you think about this implementation ? `isNotFound` is usually [defined](https://github.com/snowplow/iglu-scala-client/blob/master/modules/data/src/main/scala/com.snowplowanalytics.iglu/client/ClientError.scala#L40-L41) in `ClientError`, but there it has only access to `SortedMap[String, LookupHistory]` and it can't infer which repos are for Iglu Central, only the `Resolver`, which has the list of all `Registry`s, can ([here](https://github.com/snowplow/iglu-scala-client/blob/not_use_list/modules/core/src/main/scala/com.snowplowanalytics.iglu/client/resolver/Resolver.scala#L41)). Changing that now would be quite a big redesign and I don't think that it should happen in this ticket. I still need to add a unit test for this function.